### PR TITLE
InlineCallout component icons improvements

### LINKED ISSUE #91: Icon support
in the old docs we could have pages like this https://raw.githubusercontent.com/open-metadata/OpenMetadata/main/openmetadata-docs/content/deployment/index.md where we say:

```
<InlineCallout
    color="violet-70"
    icon="add_moderator"
    bold="Enable Security"
    href="/deployment/security"
  >
    Enable Security for your OpenMetadata deployment
  </InlineCallout>
```

And we could add this icon value with some icons from sources like material icons, or similar. I think we are now passing the icons as SVGs directly. Not sure if we can point to external icons to make it easier to try out and change

### PR DESCRIPTION
**Pull request by aniketkatkar97**
fixes #91 

- [x] Added ability to use icons from material icons in the inlineCallout component

https://user-images.githubusercontent.com/51777795/236113799-29f87ab0-3ec0-46db-88f7-be17517cfce2.mov

### CODE REVIEW

**Review by ShaileshParmar11 — APPROVED** [review]

**Review by ShaileshParmar11 — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -7,8 +7,12 @@
     "export": "next build && next export"
   },
   "dependencies": {
+    "@emotion/react": "^11.10.8",
+    "@emotion/styled": "^11.10.8",
     "@markdoc/markdoc": "latest",
     "@markdoc/next.js": "latest",
+    "@mui/icons-material": "^5.11.16",
+    "@mui/material": "^5.12.3",
```
Comment (chirag-madlani) [inline package.json:15]: @aniketkatkar97 it's a huge lib for the docs, can you check some alternatives?

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (aniketkatkar97) [reply package.json:15]: Ok @chirag-madlani, I'll check

Comment (Sachin-chaurasiya) [reply package.json:15]: @aniketkatkar97 , this might be useful https://react-icons.github.io/react-icons/

Comment (aniketkatkar97) [reply package.json]: Thanks @Sachin-chaurasiya

File: utils/CommonUtils.ts [hunk 2]
```diff
@@ -37,3 +38,11 @@ export const fetchMenuList = async (version: string) => {
     return [];
   }
 };
+
+export const getMaterialDesignIconConstructor = (iconName: string) => {
```
Comment (chirag-madlani) [inline utils/CommonUtils.ts:42]: Can we rename this to remove MaterialDesignIcon?

File: markdoc/tags/inlineCallout.markdoc.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
 export const inlineCallout = {
   render: "InlineCallout",
   attributes: {
-    icon: { type: String },
+    icon: {
+      type: String,
+      description:
+        "The identifier string to render the icons from 'Material Icons'",
```
Comment (chirag-madlani) [inline markdoc/tags/inlineCallout.markdoc.ts:7]: Are we still using Material Icons?

**Review by Sachin-chaurasiya — APPROVED** [review]
Looks Good 👍

File: utils/CommonUtils.ts [hunk 4]
```diff
@@ -37,3 +38,21 @@ export const fetchMenuList = async (version: string) => {
...
+  // i.e. 'MdCircle' for 'circle' icon
+  const regexToMatchMdPattern = /^Md/g;
+  const prefixExits = regexToMatchMdPattern.test(icon);
+
+  // If prefix doesn't exist, apply the prefix
+  const prefix = prefixExits ? "" : `Md`;
+
+  // Logic to fix the icon name passed with the incorrect cases or pattern
+  // i.e. if passed `open_lock` or `Open lock` instead of `OpenLock`
+  const iconName = startCase(icon).replaceAll(" ", "");
```
Comment (chirag-madlani) [inline utils/CommonUtils.ts:53]:
```suggestion
  const iconName = camelCase(startCase(icon));
```

File: utils/CommonUtils.ts [hunk 5]
```diff
@@ -37,3 +38,21 @@ export const fetchMenuList = async (version: string) => {
...
 };
+
+export const materialDesignIcon = (icon: string) => {
+  // Checking whether the icon name passed has prefix 'Md' applied already
+  // i.e. 'MdCircle' for 'circle' icon
+  const regexToMatchMdPattern = /^Md/g;
+  const prefixExits = regexToMatchMdPattern.test(icon);
+
+  // If prefix doesn't exist, apply the prefix
+  const prefix = prefixExits ? "" : `Md`;
```
Comment (chirag-madlani) [inline utils/CommonUtils.ts:49]:
```suggestion
if (!/^Md/g.test(icon)) {
    icon += 'Md';
  }
```

File: utils/CommonUtils.ts (hunk 4 above)
Comment (aniketkatkar97) [reply utils/CommonUtils.ts:53]: using camelCase will result in the first letter being small, we want that capital as well, hence I have used this

**Pull request merged**